src/ast.rs

```diff
@@ -40,116 +39,9 @@ lazy_static! {
     Regex::new(r#"(?i)\stypes\s*=\s*["']([^"']*)["']"#).unwrap();
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
-pub struct Position {
-  /// The 0-indexed line index.
-  pub line: usize,
-  /// The 0-indexed character index.
-  pub character: usize,
-}
-
-impl Position {
-  fn from_pos(parsed_source: &ParsedSource, pos: BytePos) -> Self {
-    let line_and_column_index =
-      parsed_source.source().line_and_column_index(pos);
-    Self {
-      line: line_and_column_index.line_index,
-      character: line_and_column_index.column_index,
-    }
-  }
-}
-
-#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
-pub struct Range {
-  pub start: Position,
-  pub end: Position,
-}
-
-impl Default for Range {
-  fn default() -> Self {
-    Self {
-      start: Position {
-        line: 0,
-        character: 0,
-      },
-      end: Position {
-        line: 0,
-        character: 0,
-      },
-    }
-  }
-}
-
-impl Range {
-  pub(crate) fn from_span(
-    parsed_source: &ParsedSource,
-    span: &deno_ast::swc::common::Span,
-  ) -> Range {
-    Range {
-      start: Position::from_pos(parsed_source, span.lo),
-      end: Position::from_pos(parsed_source, span.hi),
-    }
-  }
-
-  fn from_comment_match(
-    comment: &Comment,
-    m: &Match,
-    parsed_source: &ParsedSource,
-  ) -> Self {
-    Self {
-      start: Position::from_pos(
-        parsed_source,
-        comment.span.lo + BytePos((m.start() + 1) as u32),
```

Comment:
Should have been `+ 2` because the comment text starts after the `//`. This is fixed and tested now.